db_whitebox_test.go

```diff
@@ -112,6 +119,156 @@ func TestMethodPage(t *testing.T) {
 	}
 }
 
+func TestTx_Check_CorruptPage(t *testing.T) {
+	testCases := []struct {
+		name      string
+		bucketKey string
+		isPanic   bool
+	}{
+		{
+			name:      "corrupt page by violating btree invariant",
+			bucketKey: "testBucket",
+			isPanic:   false,
+		},
+		{
+			name:      "corrupt page by dumping random bytes",
+			bucketKey: "testBucket",
+			isPanic:   true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fileName, err := prepareData(t)
+			require.NoError(t, err)
+
+			t.Logf("Creating db file '%v'.", fileName)
+			db, err := Open(fileName, 0666, DefaultOptions)
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, db.Close())
+			}()
+
+			uErr := db.Update(func(tx *Tx) error {
+				t.Logf("Creating bucket '%v'.", tc.bucketKey)
+				b, bErr := tx.CreateBucketIfNotExists([]byte(tc.bucketKey))
+				require.NoError(t, bErr)
+				t.Logf("Generating random data in bucket '%v'.", tc.bucketKey)
+				generateSampleDataInBucket(t, b, 3)
+				return nil
+			})
+			require.NoError(t, uErr)
+
+			t.Logf("Corrupting random leaf page in bucket '%v'.", tc.bucketKey)
+			victimPageId := corruptLeafPage(t, db, tc.isPanic)
+			t.Log("Running consistency check.")
+			vErr := db.View(func(tx *Tx) error {
```

Comment:
@ahrtr as discussed I tried to use `tc.check()`, but it goes into infinite loop, i tried debugging but got no where 

the same code runs normally when using `tc.Check()`

Reply:
Can you try to figure out the root cause?

Reply:
fixed 👍🏽